src/resolvers-cassandra/Facts/queries.js

```diff
@@ -23,26 +24,24 @@ function cassandraRowToFact(row) {
   };
 }
 
-function makeDefaultClauses() {
-  const clauses = [];
-  const params = [];
-
-  clauses.push('(pipeline IN (\'TadaWeb\', \'Bing\', \'CustomEvent\'))');
-
-  return {clauses: clauses, params: params};
-}
+const SUPPORTED_PIPELINES = ['TadaWeb', 'Bing', 'CustomEvent'];
 
 function makeListQueries(args) {
-  const defaults = makeDefaultClauses();
+  return cross(args.tagFilter, SUPPORTED_PIPELINES).map(keywordAndPipeline => {
+    const clauses = [];
+    const params = [];
 
-  return args.tagFilter.map(keyword => {
-    const clauses = defaults.clauses.slice();
-    const params = defaults.params.slice();
+    if (keywordAndPipeline.a) {
+      clauses.push('(detectedkeywords CONTAINS ?)');
+      params.push(keywordAndPipeline.a);
+    }
 
-    clauses.push('(detectedkeywords CONTAINS ?)');
-    params.push(keyword);
+    if (keywordAndPipeline.b) {
+      clauses.push('(pipeline = ?)');
+      params.push(keywordAndPipeline.b);
+    }
 
-    const query = `SELECT * FROM fortis.events WHERE (${clauses.join(' AND ')})`;
+    const query = `SELECT * FROM fortis.events WHERE ${clauses.join(' AND ')} ALLOW FILTERING`;
```

Comment:
Another ouch. AFAIK, `ALLOW FILTERING` fetches everything and does driver-side filtering, right?